src/io/flutter/sdk/IsFlutterProjectCondition.java

```diff
@@ -0,0 +1,26 @@
+/*
+ * Copyright 2020 The Chromium Authors. All rights reserved.
+ * Use of this source code is governed by a BSD-style license that can be
+ * found in the LICENSE file.
+ */
+package io.flutter.sdk;
+
+import com.intellij.openapi.module.Module;
+import com.intellij.openapi.project.Project;
+import com.intellij.openapi.util.Condition;
+import io.flutter.utils.FlutterModuleUtils;
+
+import java.util.List;
+
+/**
+ * A condition that returns true if any modules include a content root with a Dart package
+ * with a 'flutter:' dependency.
+ */
+public class IsFlutterProjectCondition implements Condition<Project> {
+  @Override
+  public boolean value(Project project) {
+    final List<Module> flutterModules = FlutterModuleUtils.findModulesWithFlutterContents(project);
```

Comment:
You may want to check this with a Bazel project. I just saw a case where a bazel project didn't show up as having a Flutter module but otherwise worked fine.
WorkspaceCache.getInstance(project).isBazel()
will cheaply indicate whether a project uses bazel. The downside is it might report some bazel projects that don't actually use Dart.